pkg/nsx/services/securitypolicy/expand_test.go

```diff
@@ -236,3 +242,373 @@ func TestSecurityPolicyService_getPodSelectors(t *testing.T) {
+			namespace, _ := obj.(*core_v1.Namespace)
+			namespace.UID = types.UID("ns1-uid")
+			return nil
+		}).AnyTimes()
+	k8sClient.EXPECT().List(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).Do(
+		func(_ context.Context, objList client.ObjectList, opts ...*client.ListOptions) error {
+			podsList, _ := objList.(*core_v1.PodList)
+			podsList.Items = []core_v1.Pod{
+				{
+					Spec: core_v1.PodSpec{
+						Containers: []core_v1.Container{
+							{
+								Ports: []core_v1.ContainerPort{
+									{
+										Name:          "http",
+										Protocol:      "TCP",
+										ContainerPort: 8080,
+									},
+								},
+							},
+						},
+					},
+					Status: core_v1.PodStatus{
+						Phase: core_v1.PodRunning,
+						PodIP: "1.1.1.1",
+					},
+				},
+			}
+			return nil
+		},
+	).AnyTimes()
+
+	mockVPCService := common.MockVPCServiceProvider{VPCResource: &common.VPCResourceInfo{
+		OrgID:     "default",
+		ProjectID: "pro1",
+		VPCID:     "vpc1",
+	}}
+	mockVPCService.On("ListVPCInfo", mock.Anything).Return(nil).Maybe()
+
+	getTestIPsetGroup := func(id string, displayName string, ruleId string, isVPC bool, policyType string) *model.Group {
+		groupTags := []model.Tag{
+			{Scope: common.String("nsx-op/group_type"), Tag: common.String("destination")},
+			{Scope: common.String("nsx-op/rule_id"), Tag: common.String(ruleId)},
+			{Scope: common.String("nsx-op/selector_hash"), Tag: common.String("2be88ca4242c76e8253ac62474851065032d6833")},
+		}
+		groupTags = append(groupTags, ruleTagsFn(policyType)...)
+		if isVPC {
+			groupTags = append(groupTags,
+				model.Tag{Scope: common.String("nsx-op/nsx_share_created_for"), Tag: common.String("notShared")})
+		}
+
+		addresses := data.NewListValue()
+		addresses.Add(data.NewStringValue("1.1.1.1"))
+
+		return &model.Group{
+			Id:          common.String(id),
+			DisplayName: common.String(displayName),
+			Tags:        groupTags,
+			Expression: []*data.StructValue{
+				data.NewStructValue(
+					"",
+					map[string]data.DataValue{
+						"resource_type": data.NewStringValue("IPAddressExpression"),
+						"ip_addresses":  addresses,
+					},
+				),
+			},
+		}
+	}
+
+	for _, tc := range []struct {
+		name       string
+		vpcEnabled bool
+		ruleIdx    int
+		createdFor string
+		expGroups  []*model.Group
+		expRules   []*model.Rule
+		expErr     string
+	}{
+		{
+			name:       "VPC: rule without and ports for NetworkPolicy",
+			vpcEnabled: true,
+			ruleIdx:    0,
+			createdFor: common.ResourceTypeNetworkPolicy,
+			expRules: []*model.Rule{
+				{
+					Id:             common.String("p1_uid1_26e848bd_all"),
+					DisplayName:    common.String("rule1"),
+					Direction:      common.String(string("IN")),
+					SequenceNumber: Int64(int64(0)),
+					Action:         common.String(string("ALLOW")),
+					Services:       []string{"ANY"},
+					Tags:           npRuleTags,
+				},
+			},
+		}, {
+			name:       "T1: rule without and ports for NetworkPolicy",
```

Comment:
For T1, we don't support networkpolicy, I think we can remove networkpolicy cases for T1 to avoid confuse.